server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -105,303 +105,305 @@ class AddressIndexRepository @Inject()(
           field = "paf.dependentLocality",
           value = token
-        ).boost(queryParams.locality.pafDependentLocalityBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.locality.pafDependentLocalityBoost)),
       tokens.get(Tokens.locality).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "paf.welshDependentLocality",
           value = token
-        ).boost(queryParams.locality.pafWelshDependentLocalityBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.locality.pafWelshDependentLocalityBoost)),
       tokens.get(Tokens.locality).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.locality",
           value = token
-        ).boost(queryParams.locality.lpiLocalityBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.locality.lpiLocalityBoost)),
       tokens.get(Tokens.locality).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "paf.doubleDependentLocality",
           value = token
-        ).boost(queryParams.locality.pafDoubleDependentLocalityBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.locality.pafDoubleDependentLocalityBoost)),
       tokens.get(Tokens.locality).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "paf.welshDoubleDependentLocality",
           value = token
-        ).boost(queryParams.locality.pafWelshDoubleDependentLocalityBoost))
-    ).flatten.map(_.fuzziness("1"))
-
-    val allQuery = Seq(
-      matchQuery(
-        field = "paf.pafAll",
-        value = originalInput.toUpperCase
-      ).boost(queryParams.pafAllBoost),
-      matchQuery(
-        field = "lpi.nagAll",
-        value = originalInput.toUpperCase
-      ).boost(queryParams.nagAllBoost)
-    )
-
-    val preciseQuery = Seq(
+        ).fuzziness(defaultFuzziness)).boost(queryParams.locality.pafWelshDoubleDependentLocalityBoost))
+    ).flatten
+
+    val allQuery =
+      moreLikeThisQuery(Seq("paf.pafAll", "lpi.nagAll"))
+        .like(originalInput.toUpperCase)
+        .minTermFreq(1)
+        .analyser("welsh_split_analyzer")
+
+    // minimumShouldMatch method does not exits for moreLikeThisQuery. This a mutation (side effect) of the code above
+    allQuery.builder.minimumShouldMatch("-1")
+
+    val preciseMustQuery = Seq(
       buildingNumberQuery,
       buildingNameQuery,
       subBuildingNameQuery,
       streetNameQuery,
       townNameQuery,
       postcodeQuery,
-      postcodeInOutQuery,
       paoQuery,
-      saoQuery,
+      saoQuery
+      // `dismax` dsl does not exist, `: _*` means that we provide a list (`queries`) as arguments (args) for the function
+    ).filter(_.nonEmpty).map(queries => dismax.query(queries: _*).tieBreaker(queryParams.disMaxTieBreaker))
+
+    val preciseShouldQuery = Seq(
       organisationNameQuery,
       departmentNameQuery,
       localityQuery
```

Comment:
Fine for now, but after results are known, we should test if changing this to similar as the townNameQuery and moving it inside the preciseMustQuery helps to limit some unwanted matches.